apps/web/src/services/asset-inventory/collector/CollectorPage.vue

```diff
@@ -148,31 +129,39 @@ const collectorApiQueryHelper = new ApiQueryHelper()
         COLLECTOR_QUERY_HELPER_SET.PLUGIN_INFO,
         COLLECTOR_QUERY_HELPER_SET.STATE,
     )
-    .setPage(state.pageStart, state.pageLimit)
-    .setSort(state.sortBy, true);
+    .setPage(collectorPageState.pageStart, collectorPageState.pageLimit)
+    .setSort(collectorPageState.sortBy, true);
 
-/* API */
-const getCollectorList = async () => {
-    collectorApiQueryHelper.setFilters(cloudCollectorPageStore.allFilters);
+/* Components */
+const initCollectorList = async () => {
+    collectorApiQueryHelper.setFilters(collectorPageStore.allFilters);
     try {
-        const res = await SpaceConnector.client.inventory.collector.list({
-            query: collectorApiQueryHelper.data,
-        });
-        state.collectors = res.results;
-        state.totalCount = res.total_count;
+        await collectorPageStore.getCollectorList(collectorApiQueryHelper.data);
+        if (Object.keys(state.items).length > 0) {
+            await collectorPageStore.setCollectorList(state.items);
+        }
     } catch (e) {
         ErrorHandler.handleError(e);
+        await collectorPageStore.$reset();
     }
 };
+const handleChangeToolbox = async (options) => {
+    setApiQueryWithToolboxOptions(collectorApiQueryHelper, options);
+    await initCollectorList();
+};
 
 /* Watcher */
-watch(() => cloudCollectorPageState.selectedProvider, async () => {
-    await filterByProvider();
+watch(() => collectorPageState.selectedProvider, async () => {
+    await initCollectorList();
+});
+
+/* Unmounted */
+onUnmounted(() => {
+    collectorPageStore.$reset();
```

Comment:
```suggestion
collectorPageStore.$dispose();
```
suggestion: I recommend using `$dispose()` because it completely removes the store. You can also confirm its complete removal in the dev tools.
(`$reset()` is included in`$dispose`)